saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/assertion/SAMLProxyRestrictionParser.java

```diff
@@ -0,0 +1,45 @@
+package org.keycloak.saml.processing.core.parsers.saml.assertion;
+
+import org.keycloak.dom.saml.v2.assertion.ProxyRestrictionType;
+import org.keycloak.saml.common.exceptions.ParsingException;
+import org.keycloak.saml.common.util.StaxParserUtil;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.events.StartElement;
+import java.math.BigInteger;
+
+/**
+ * Parse the <ProxyRestriction Count=\"\"> tag
+ *
+ * @author Patric Vormstein
+ * @since 21.03.2018
+ */
+public class SAMLProxyRestrictionParser extends AbstractStaxSamlAssertionParser<ProxyRestrictionType> {
+
+    private static final SAMLProxyRestrictionParser INSTANCE = new SAMLProxyRestrictionParser();
+
+    public SAMLProxyRestrictionParser() {
+        super(SAMLAssertionQNames.PROXY_RESTRICTION);
+    }
+
+    public static SAMLProxyRestrictionParser getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    protected ProxyRestrictionType instantiateElement(XMLEventReader xmlEventReader, StartElement element) throws ParsingException {
+        ProxyRestrictionType proxyRestriction = new ProxyRestrictionType();
+        Integer count = StaxParserUtil.getIntegerAttributeValue(element, SAMLAssertionQNames.ATTR_COUNT);
+        proxyRestriction.setCount(BigInteger.valueOf(count));
```

Comment:
This throws NPE if `count == null` since `Count` attribute is optional. Needs to be guarded by a condition.